### PR TITLE
[Merged by Bors] - [Merged by Bors] - feat: condensed sets are equivalent to sheaves on `Stonean` and `Profinite`

### PR DESCRIPTION
**Pull request by dagurtomas**
Co-authored-by: Adam Topaz @adamtopaz 
Co-authored-by: Nick Kuhn @nick-kuhn 

- [x] depends on: #6809 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by ADedecker** [comment]
Is this ready for review?

File: Mathlib/Condensed/Equivalence.lean [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+lemma lift_lifts {X Y : Profinite} {Z : Stonean} (e : Stonean.toProfinite.obj Z ⟶ Y) (f : X ⟶ Y)
+    [Epi f] : lift e f ≫ f = e := CompHaus.lift_lifts _ _
+
+lemma projective_of_extrDisc {X : Profinite} (hX : ExtremallyDisconnected X) :
+    Projective X:= by
+  let X' : Stonean := ⟨X.toCompHaus⟩
+  show Projective (Stonean.toProfinite.obj X')
+  exact inferInstance
+
+end Profinite
```
Comment (adamtopaz) [inline Mathlib/Condensed/Equivalence.lean:209]: This stuff is now all part of `Topology/Category/Stonean/Basic`.

File: Mathlib/Condensed/Equivalence.lean [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+  convert generate_singleton_mem_coherentTopology B
+  ext Y f
+  refine ⟨fun ⟨⟨obj, lift, map, fact⟩⟩ ↦ ?_, fun ⟨Z, h, g, hypo1, hf⟩ ↦ ?_⟩
+  · obtain ⟨p, p_factors⟩ := Projective.factors map (Profinite.presentation.π B)
+    exact ⟨(Stonean.toProfinite.obj (presentation B)), ⟨lift ≫ p, ⟨(presentation.π B),
+      ⟨Presieve.singleton.mk, by rw [Category.assoc, p_factors, fact]⟩⟩⟩⟩
+  · cases hypo1
+    exact ⟨⟨presentation B, h, presentation.π B, hf⟩⟩
+
+theorem coverDense.inducedTopology_Sieve_iff_EffectiveEpiFamily (X : Stonean) (S : Sieve X) :
```
Comment (adamtopaz) [inline Mathlib/Condensed/Equivalence.lean:249]: If this really needed? I had the impression that we now had this for the coherent topology on any precoherent category.

**Review by adamtopaz — CHANGES_REQUESTED** [review]

File: Mathlib/Condensed/Equivalence.lean (hunk 1 above)
Comment (dagurtomas) [reply Mathlib/Condensed/Equivalence.lean:209]: Oh sorry I didn't merge master after #6809 was merged.

File: Mathlib/Condensed/Equivalence.lean (hunk 2 above)
Comment (erdOne) [reply Mathlib/Condensed/Equivalence.lean:170]: I think this lemma together with `coherentTopology.mem_sieves_iff_hasEffectiveEpiFamily` that you mentioned shows that the induced topology coincides with the coherent topology.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**